data-structure/jqueue.php

```diff
@@ -0,0 +1,71 @@
+<?php
+class jQueue extends jDataStructures
+{
+    public function __construct($data)
+    {
+        parent::__construct($data);
+    }
+
+    /**
+     * put the element to the first position
+     * @param mixed $elem
+     * @return jQueue
+     */
+    public function push($elem)
+    {
+        array_unshift($this->_data, $elem);
```

Comment:
why are you using `array_unshift`? new elements should get added to the end of the queue. Shouldn't you use `array_push` instead?